### PR TITLE
Drafted API for dataset updates schedule operations

### LINKED ISSUE #280: Introduce a component to store dataset flows schedule settings
A logical component, that allows reading and saving flow scheduling instructions for each dataset:
- any dataset: flows enabled/disabled
- support API for all dataset flows kinds
- for updates in particular:
   - root polling source: ingest job - frequency of polling (CRON vs time delta)
   - derived: execute query job -  throttling parameters (immediate update, minimal delay after first or last update, minimal data range to accumulate)
   - Future - root with push source - similar throttling parameters
   - all dataset kinds - compacting job frequency
- flows are disabled by default in the model

Storage:
- In-memory - to use in CLI or API server mode, empty after restart, everything resets to defaults
- Future: durable database-backed implementation in platform

GraphQL API:
- read dataset flow rules (Reader permission)
- update dataset flow rules (Writer permission)

TODO: separate ticket - similar responsibilities for System flows, accessible only for admins.

### PR DESCRIPTION
**Pull request by zaychenko-sergei**
Closes #280

### CODE REVIEW

File: src/adapter/graphql/src/mutations/dataset_updates_schedule_mut.rs [hunk 1]
```diff
@@ -0,0 +1,251 @@
...
+        self.apply_pause_resume_schedule(update_schedule_service.as_ref(), paused, &update_state)
+            .await?;
+
+        Ok(SetUpdateScheduleResult::Success(
+            SetUpdateScheduleResultSuccess { success: true },
+        ))
+    }
+
+    #[graphql(guard = "LoggedInGuard::new()")]
+    async fn set_throttling(
```
Comment (sergiimk) [inline src/adapter/graphql/src/mutations/dataset_updates_schedule_mut.rs:99]: Name `set_batching` seems more appropriate to me - as we define how to combine high-frequency updates into batches.

I would actually rename `throttling_period` here to `batching_period` as I think we should use the term `throttling` purely for additional artificial processing delay we will be adding on public platform to save costs.

**Review by sergiimk — CHANGES_REQUESTED** [review]

File: resources/schema.gql [hunk 2]
```diff
@@ -868,6 +904,15 @@ type SetTransform {
 	transform: Transform!
 }
 
+interface SetUpdateScheduleResult {
+	message: String!
+}
+
+type SetUpdateScheduleResultSuccess implements SetUpdateScheduleResult {
+	success: Boolean!
```
[the remark this answers is not in the document]
Comment (zaychenko-sergei) [reply resources/schema.gql]: I simplified to `Result<bool>`

File: src/adapter/graphql/src/mutations/dataset_updates_schedule_mut.rs (hunk 1 above)
Comment (zaychenko-sergei) [reply src/adapter/graphql/src/mutations/dataset_updates_schedule_mut.rs:91]: We are using term "reactive" schedule, as a contrast to "proactive", so my name of preference now is `set_reactive`

File: resources/schema.gql [hunk 3]
```diff
@@ -374,6 +386,30 @@ scalar DatasetRef
 
 scalar DatasetRefAny
 
+union DatasetUpdatesPollingSchedule = TimeDelta | CronExpression
+
+type DatasetUpdatesSchedule {
+	paused: Boolean!
+	updateSettings: DatasetUpdatesSettings!
```
[the remark this answers is not in the document]
Comment (zaychenko-sergei) [reply resources/schema.gql]: Implemented similar to your suggestion. Although, what I don't like about it,is that this type has a possible invalid state, when all optional parts are `None`. It should not happen at run-time, but is possible by design, unlike my non-idiomatic version :-)

**Comment by zaychenko-sergei** [comment]
This branch is no longer mergeable, and with flow system taken different shape, will have to be restarted.

**Pull request closed without merging**